frontend/src/modules/calculator/containers/Keyboard.js

```diff
@@ -0,0 +1,266 @@
+                                else
+                                    dispatch(
+                                        setError(
+                                            'Количество открытых скобок должно быть больше количества закрывающих скобок'
+                                        )
+                                    );
+                            }}
+                        >
+                            )
+                        </FunctionKey>
+                    </CustomCell>
+                    <CustomCell>
+                        <FunctionKey onClick={() => addExpression('%')}>%</FunctionKey>
+                    </CustomCell>
+                    <CustomCell>
+                        {currentExpression.length > 1 ? (
+                            <ClearKey onClick={clearEntryExpression} isCE />
+                        ) : (
+                            <ClearKey onClick={allClearExpression} />
+                        )}
+                    </CustomCell>
+                </TableRow>
+                <TableRow>
+                    <CustomCell>
+                        <InvKey onClick={switchIsInv} isInv={isInv}>
+                            Inv
+                        </InvKey>
+                    </CustomCell>
+                    <CustomCell>
+                        {isInv ? (
+                            <FunctionKey onClick={() => addExpression('sin<sup>-1</sup>(')}>
+                                sin<CustomSup>-1</CustomSup>
+                            </FunctionKey>
+                        ) : (
+                            <FunctionKey onClick={() => addExpression('sin(')}>sin</FunctionKey>
+                        )}
+                    </CustomCell>
+                    <CustomCell>
+                        {isInv ? (
+                            <FunctionKey onClick={() => addExpression('e<sup>')}>
+                                e<CustomSup>x</CustomSup>
+                            </FunctionKey>
+                        ) : (
+                            <FunctionKey onClick={() => addExpression('ln(')}>ln</FunctionKey>
+                        )}
+                    </CustomCell>
+                    <CustomCell>
+                        <NumberKey onClick={() => addExpression('7')}>7</NumberKey>
+                    </CustomCell>
+                    <CustomCell>
+                        <NumberKey onClick={() => addExpression('8')}>8</NumberKey>
+                    </CustomCell>
+                    <CustomCell>
+                        <NumberKey onClick={() => addExpression('9')}>9</NumberKey>
+                    </CustomCell>
+                    <CustomCell>
+                        <OperationKey onClick={() => addExpression('÷')}>÷</OperationKey>
+                    </CustomCell>
+                </TableRow>
+                <TableRow>
+                    <CustomCell>
+                        <FunctionKey onClick={() => addExpression('π ')}>π</FunctionKey>
```

Comment:
@thienlongtpct In my original task I didn't ask you about this functionality, but, ok, cool